libr/core/core.c

```diff
@@ -1211,7 +1211,8 @@ R_API int r_core_serve(RCore *core, RIODesc *file) {
 			return -1;
 		if (c == NULL) {
 			eprintf ("rap: cannot accept\n");
-			r_socket_close (c);
+			/*r_socket_close (c);*/
+			r_socket_free (c);
```

Comment:
Should be always `r_socket_free` or `r_socket_close` ? r_socket_close doesn't free the RSock structure and I understand that if it fails the structure should be freed